mr/src/main/java/org/elasticsearch/hadoop/serialization/dto/IndicesAliases.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.serialization.dto;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IndicesAliases {
+    private Map<String, Map<String, Alias> > indices;
+
+    private IndicesAliases(Map<String, Map<String, Alias> > indices) {
+        this.indices = indices;
+    }
+
+    public Map<String, Alias> getAliases(String index) {
+        return indices.get(index);
+    }
+
+    public static IndicesAliases parse(Map<String, Object> resp) {
+        final Map<String, Map<String, Alias> > indices = new HashMap<String, Map<String, Alias> > ();
+        for (Map.Entry<String, Object> index : resp.entrySet()) {
+            final Map<String, Object> metadata = (Map<String, Object>) index.getValue();
+            final Map<String, Object> aliases = (Map<String, Object>) metadata.get("aliases");
+            final Map<String, Alias> indexAliases = new HashMap<String, Alias> ();
+            indices.put(index.getKey(), indexAliases);
+            for (Map.Entry<String, Object> entry : aliases.entrySet()) {
+                String name = entry.getKey();
+                String searchRouting = null;
+                String indexRouting = null;
+                Map<String, Object> filter = null;
+
+                if (metadata.containsKey("search_routing")) {
+                    searchRouting = (String) metadata.get("search_routing");
+                }
+                if (metadata.containsKey("index_routing")) {
+                    indexRouting = (String) metadata.get("index_routing");
+                }
+                if (metadata.containsKey("filter")) {
+                    filter = (Map<String, Object>) metadata.get("filter");
+                }
```

Comment:
Should these be `((Map<String, Object>)value).containsKey("...")`? The `metadata` map is only ever going to have one value named `aliases`.